UNITTESTS/stubs/stoip4_stub.c

```diff
@@ -24,11 +24,55 @@
 /**
  * Convert numeric IPv4 address string to a binary.
  * \param ip4addr IPv4 address in string format.
- * \param len Length of IPv4 string, maximum of 16..
+ * \param len Length of IPv4 string.
  * \param dest buffer for address. MUST be 4 bytes.
  * \return boolean set to true if conversion succeded, false if it didn't
  */
 bool stoip4(const char *ip4addr, size_t len, void *dest)
 {
+
```

Comment:
Instead of copying the `stoip4()` function here, can we modify the build scripts to include the original code.

Maintaining a stub and original, which both are identical, does not make sense.

Reply:
It is not exactly the original. It never returns false and doesn't even perform all the checks, so it can be safely used in other tests, which don't really care about the data being copied or not, and would do fine with an empty stub.
However, to check that the `getpeername` works we need to perform the copy in this one place - hence the copying algorithm runs if the inputs are valid.
If we were to use the actual function we would have to guarantee the input parameters validity. Should we?

Reply:
If I remove the usage of this stub from TLSSocket tests, it passes OK.
```
$ git diff
diff --git i/UNITTESTS/features/netsocket/TLSSocketWrapper/unittest.cmake w/UNITTESTS/features/netsocket/TLSSocketWrapper/unittest.cmake
index 2ff13c2dc6..d126328baf 100644
--- i/UNITTESTS/features/netsocket/TLSSocketWrapper/unittest.cmake
+++ w/UNITTESTS/features/netsocket/TLSSocketWrapper/unittest.cmake
@@ -28,8 +28,6 @@ set(unittest-test-sources
   stubs/mbed_shared_queues_stub.cpp
   stubs/nsapi_dns_stub.cpp
   stubs/EventFlags_stub.cpp
-  stubs/stoip4_stub.c
-  stubs/ip4tos_stub.c
   stubs/SocketStats_Stub.cpp
 )
```

Notice that `UNITTESTS/features/netsocket/TLSSocketWrapper/unittest.cmake` already seem to include the original `stoip4()`

I does not seem to require `getpeername()`, that is tested elsewhere.

Reply:
Thanks, good point. We haven't tried it, but it works.